File: app/src/main/java/com/example/calculatorfragment/MainActivity.java
package com.example.calculatorfragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.ListFragment;

import android.os.Bundle;
import android.util.DisplayMetrics;

public class MainActivity extends AppCompatActivity implements ButtonFragment.OnFragmentSendDataListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DisplayMetrics metrics = new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getMetrics(metrics);
        int displayWidth =  metrics.widthPixels;
        int displayHeight =  metrics.heightPixels;
    }

    @Override
    public void onSendData(String selectedItem) {
        JournalFragment fragment = (JournalFragment) getSupportFragmentManager()
                .findFragmentById(R.id.fragment_container_view);
        if (fragment != null)
            fragment.setSelectedItem(selectedItem);
    }
}

File: app/src/main/java/com/example/calculatorfragment/MainActivity.kt
package com.example.calculatorfragment

import androidx.appcompat.app.AppCompatActivity
import com.example.calculatorfragment.ButtonFragment.OnFragmentSendDataListener
import android.os.Bundle
import java.lang.StringBuilder

class MainActivity : AppCompatActivity(), OnFragmentSendDataListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onSendData(selectedItem: StringBuilder?) {
        val fragment = supportFragmentManager
            .findFragmentById(R.id.fragment_container_view) as JournalFragment?
        fragment?.setSelectedItem(selectedItem)
    }
}